integration/feature/init/src/MillInitSbtTests.scala

```diff
@@ -0,0 +1,332 @@
+  def tests: Tests = Tests {
+    /*
+    - 28 KB
+    - originally sbt 1.2.8
+     */
+    val url = "https://github.com/tototoshi/scala-csv/archive/refs/tags/1.3.6.zip"
+
+    test - integrationTest(url) { tester =>
+      bumpSbtTo1107(tester.workspacePath)
+
+      tester.testMillInit(
+        expectedCompileTasks = Some(SplitResolvedTasks(
+          Seq("compile", "test.compile"),
+          Seq.empty
+        )),
+        expectedTestTasks = Some(SplitResolvedTasks(
+          Seq(),
+          /*
+          Paths relative to the workspace are used in the test sources such as `new File("src/test/resources/simple.csv")`
+          and they seem to cause the test to fail with Mill:
+          ```text
+          java.io.FileNotFoundException: src/test/resources/simple.csv (No such file or directory)
+          ```
+           */
+          Seq("test")
+        ))
+      )
+    }
+  }
+}
+
+// same as the one in the unit tests
+object MillInitSbtMultiProjectExampleTests extends BuildGenTestSuite {
+  def tests: Tests = Tests {
+    /*
+    - 12 KB
+    - originally sbt 1.0.2
+     */
+    val url =
+      "https://github.com/pbassiner/sbt-multi-project-example/archive/152b31df9837115b183576b0080628b43c505389.zip"
+
+    test - integrationTest(url) { tester =>
+      bumpSbtTo1107(tester.workspacePath)
+
+      val submodules = Seq("common", "multi1", "multi2")
+      if (System.getProperty("java.version").split('.').head.toInt <= 11)
+        tester.testMillInit(
+          expectedCompileTasks = Some(SplitResolvedTasks(
+            Seq("compile") ++ submodules.flatMap(_.allCompileTasks),
+            Seq.empty
+          )),
+          expectedTestTasks = Some(SplitResolvedTasks(submodules.map(_.testTask), Seq.empty))
+        )
+      else
+        tester.testMillInit(
+          // initCommand = defaultInitCommand ++ Seq("--jvm-id", "11"),
+          expectedCompileTasks = Some({
+            /*
+            `multi1.compile` doesn't work well when Mill is run with JDK 17 and 21:
+            ```text
+            1 tasks failed
+            multi1.compile java.io.IOError: java.lang.RuntimeException: /packages cannot be represented as URI
+                java.base/jdk.internal.jrtfs.JrtPath.toUri(JrtPath.java:175)
+                scala.tools.nsc.classpath.JrtClassPath.asURLs(DirectoryClassPath.scala:183)
+                ...
+            ```
+            Passing a `jvmId` 11 doesn't work here.
+             */
+            val succeededSubmoduleCompileTasks = Seq("common.compile", "multi2.compile")
+            SplitResolvedTasks(
+              Seq("compile") ++ succeededSubmoduleCompileTasks,
+              (submodules.flatMap(_.allCompileTasks).toSet -- succeededSubmoduleCompileTasks).toSeq
+            )
+          }),
+          expectedTestTasks = Some(SplitResolvedTasks(Seq.empty, submodules.map(_.testTask)))
+        )
+    }
+  }
+}
+
+// relatively large libraries
+
+object MillInitSbtZioHttpTests extends BuildGenTestSuite {
```

Comment:
Let's drop most of these large libraries and just pick 1 large library and try to make it work. 

Apart from the big backend frameworks you've chosen here which probably are very sophisticated, how about some of the other projects listed on https://github.com/EvanLi/Github-Ranking/blob/master/Top100/Scala.md?
https://github.com/gatling/gatling is one we used in our case study (https://mill-build.org/mill/comparisons/sbt.html) that may have a simpler build configuration since it's an application rather than a core framework. https://github.com/lichess-org/lila could be worth a try as well.

Again, I don't expect to have everything compiling and all tests passing, but would be nice to get at least one (1) single test passing to verify that this conversion works at all!

Reply:
Let's just remove them

Reply:
I added a `MillInitSbtGatlingTests` as requested. Some modules `compile` and `test` tasks run successfully on my machine, while others fail because the sbt configuration "compile->compile;test->test" is not supported yet. Shall I try to support this?

Reply:
Yes let's try to support `"compile->compile;test->test"`. It means that both the `foo` depends on `bar` and also `foo.test` depends on `bar.test`. Should be straightforward to implement using `moduleDeps` on the main and test modules